src/release/calendar.js

```diff
@@ -4,28 +4,46 @@ import fetchText from '../fetch/text';
 import { parse } from '../meta/version';
 
 export default async function getCalendar({
-  channel = 'release',
+  channel = 'release', days = 0,
 } = {}) {
   const url = 'https://calendar.google.com/calendar/ical/mozilla.com_2d37383433353432352d3939%40resource.calendar.google.com/public/basic.ics';
   const ics = await fetchText(url);
   const parsed = ical.parseICS(ics);
   const dates = Object.keys(parsed).reduce((data, key) => {
     const entry = parsed[key];
-    if (moment().diff(entry.start, 'days') >= 0) {
+    let regex;
+    let version;
+    let ch;
+    // skip reference to Aurora due to merge into Beta to avoid duplicate entries
+    if (moment().diff(entry.start, 'days') >= days || entry.summary === 'MERGE: B58, A59, N60') {
       return data;
     }
-    const summary = entry.summary.match(/firefox\s+(esr)?\s*([\d.]+)\s+release/i);
+    if (channel === 'release') {
+      regex = /firefox\s+(esr)?\s*([\d.]+)\s+release/i;
+    } else if (channel === 'nightly') {
+      regex = /N[0-9]*/;
+    } else if (channel === 'beta') {
+      regex = /B[0-9]*/;
+    }
+    const summary = entry.summary.match(regex);
+
     if (!summary) {
       return data;
     }
-    const ch = summary[1] ? 'esr' : 'release';
-    if (channel && ch !== channel) {
-      return data;
+
+    if (channel === 'release') {
+      ch = summary[1] ? 'esr' : 'release';
```

Comment:
Could you add a comment here explaining what the logic is here? I have to confess I don't really understand what's going on.

Reply:
I did some refactoring on this - let me know if it doesn't make more sense.